### DISCUSSION TITLE
[Solved] Connect an N-size port to a 1 size port

### DISCUSSION THREAD

**Discussion by denis-jidovtseff-jc** [OpenModelica/OpenModelica#10027] [Q&A] [answered]
Hello, 
I'm trying to connect a connector with size N to a simple connector.
my fist idea was to use
`connect(in1[1:n],out1);` 
I have the following error : `The type of variables in1[1:n] and out1 are inconsistent in connect equations.`
if I do a loop, to mach a similar case with two different connectors connected to a signle one:
```
for i in 1:n loop
    connect(in1[i],out1);
  end for;
```
I have a different number of variable and equations and I can't solve my model

Someone has an idea to solve that?

**Comment by casella** [comment, marked as answer]
Is in1 an array of inputs and out1 an output connector? In that case, the for loop is the correct way to handle it, since the two connectors in a connect statement must have the same size. Not sure why you get an error. Are the components that you are connecting balanced? What output do you get when you check them separately?

**Reply by denis-jidovtseff-jc** [reply]
Yes the in1 is the array of input. I finally found the bug.
The integer i was not properly initialized
